jetty-server/src/main/java/org/eclipse/jetty/server/MultiPartFormInputStream.java

```diff
@@ -535,20 +541,20 @@ protected void parse()
         {
             // sort out the location to which to write the files
             if (_config.getLocation() == null)
-                _tmpDir = _contextTmpDir;
+                _tmpDir = _contextTmpDir.toPath();
             else if ("".equals(_config.getLocation()))
-                _tmpDir = _contextTmpDir;
+                _tmpDir = _contextTmpDir.toPath();
             else
             {
-                File f = new File(_config.getLocation());
-                if (f.isAbsolute())
-                    _tmpDir = f;
+                Path location = FileSystems.getDefault().getPath(_config.getLocation());
+                if (location.isAbsolute())
+                    _tmpDir = location;
                 else
-                    _tmpDir = new File(_contextTmpDir, _config.getLocation());
+                    _tmpDir = _contextTmpDir.toPath().resolve(location);
```

Comment:
This is the only relevant part of this change.  

The rest is just confusing.

Reply:
Not sure where's the confusion. Around lines 538 in the parse() method we work out where the tmp directory is so we can write to it if we need to (eg the number of bytes received exceed the configured limit for bytes retained in memory). The location of the tmp directory can be given by the MultiPartConfigElement.location, or, if that has not been configured, I default to the context tmp dir. This code has always been there, I just changed to calculating it as a Path instead of a File.

Reply:
To be clear, the main change is at lines 196-199 where we treat the String argument to Part.write() as either absolute or relative: previously we always treated it as relative.